### PR TITLE
MapView: Copy commit message action

### PR DESCRIPTION
**Pull request by lolgear**
- [x] Copy selected commit message has been added.
- [ ] Roughly #302

### CODE REVIEW

File: GitUpKit/Views/GIMapViewController.m [hunk 1]
```diff
@@ -916,6 +916,11 @@ - (IBAction)editSelectedCommitMessage:(id)sender {
   [self editCommitMessage:commit];
 }
 
+- (IBAction)copySelectedCommitMesssage:(id)sender {
```
Comment (lucasderraugh) [inline GitUpKit/Views/GIMapViewController.m]: `Messsage` -> `Message`

Comment (lolgear) [reply GitUpKit/Views/GIMapViewController.m]: Fixed :)

File: GitUpKit/Views/Base.lproj/GIMapViewController.xib [hunk 2]
```diff
@@ -111,6 +111,12 @@ DQ
                         <action selector="editSelectedCommitMessage:" target="-2" id="PrK-v7-e6I"/>
                     </connections>
                 </menuItem>
+                <menuItem title="Copy Message…" alternate="YES" keyEquivalent="e" id="LnA-LI-dEb">
```
Comment (lucasderraugh) [inline GitUpKit/Views/Base.lproj/GIMapViewController.xib]: `...` is reserved for actions where there will be something the user has to follow up on. There is no followup dialog here so the text should simply be `Copy Message`.

File: GitUpKit/Views/GIMapViewController+Operations.m [hunk 3]
```diff
@@ -395,6 +395,12 @@ - (void)editCommitMessage:(GCHistoryCommit*)commit {
                           }];
 }
 
+- (void)copyCommitMessage:(GCHistoryCommit*)commit {
+  NSString* message = _CleanedUpCommitMessage(commit.message);
+  [self.windowController showOverlayWithStyle:kGIOverlayStyle_Informational format:NSLocalizedString(@"Commit message copied: \"%@\"", nil), message];
+  [[NSPasteboard generalPasteboard] setString:message forType:NSPasteboardTypeString];
```
Comment (lucasderraugh) [inline GitUpKit/Views/GIMapViewController+Operations.m]: You don't have any `-[NSPasteboard declareTypes:owner:]` call. It's unlikely to be copying anything to the pasteboard in this case. From the docs: "The type must have been declared by a previous declareTypes:owner: message."

Comment (lolgear) [reply GitUpKit/Views/GIMapViewController+Operations.m]: @lucasderraugh Heh, my bad. It is ridiculous how different macOS and iOS at some points. Yes, all applications in iOS are sandboxed, so, they can't harm user.

Mmm, I've found a bug in current release app. I think that "testing" code by others is one of the best practices to find bugs.

In this PR I was in hurry, yes. It is one of the core functions that I need ( besides drag & drop, history of file, search in commit view  and search in selected range of commits ).

**Comment by lucasderraugh** [comment]
Very nice, looks great!

**Pull request merged**